test/hummingbot/connector/exchange/binance/test_binance_user_stream_data_source.py

```diff
@@ -177,3 +240,30 @@ def test_listen_for_user_stream_no_listen_key(self, mock_api, mock_next_tick, mo
 
         msg = self.ev_loop.run_until_complete(msg_queue.get())
         self.assertTrue(msg, self._user_update_event)
+
+    @aioresponses()
+    @patch("aiohttp.ClientSession.ws_connect", new_callable=AsyncMock)
+    def test_listen_for_user_stream_iter_message_throws_exception(self, mock_api, mock_ws):
+        url = utils.private_rest_url(path_url=CONSTANTS.BINANCE_USER_STREAM_PATH_URL, domain=self.domain)
+        regex_url = re.compile(f"^{url}".replace(".", r"\.").replace("?", r"\?"))
+
+        mock_response = {
+            "listenKey": self.listen_key
+        }
+        mock_api.post(regex_url, body=ujson.dumps(mock_response))
+
+        msg_queue: asyncio.Queue = asyncio.Queue()
+        mock_ws.return_value = self.mocking_assistant.create_websocket_mock()
+        mock_ws.return_value.receive.side_effect = lambda: self._raise_exception(Exception("TEST ERROR"))
+        mock_ws.close.return_value = None
+
+        self.listening_task = self.ev_loop.create_task(
+            self.data_source.listen_for_user_stream(self.ev_loop, msg_queue)
+        )
+
+        with self.assertRaises(asyncio.TimeoutError):
+            self.ev_loop.run_until_complete(
+                asyncio.wait_for(self.listening_task, 1)
```

Comment:
For this one, we can use the same approach as `NdaxExchangeTests._create_exception_and_unlock_test_with_event` to shave off the second